src/Namshi/JOSE/JWS.php

```diff
@@ -157,12 +159,43 @@ public function verify($key, $algo = null)
         }
 
         $decodedSignature = $this->encoder->decode($this->getEncodedSignature());
-        $signinInput = $this->generateSigninInput();
+        $signinInput = $this->getSigninInput();
 
         return $this->getSigner()->verify($key, $decodedSignature, $signinInput);
     }
 
     /**
+     * Get the original token signin input if it exists, otherwise generate the
+     * signin input for the current JWS
+     *
+     * @return string
+     */
+    private function getSigninInput()
+    {
+        $parts = explode('.', $this->originalToken);
+
+        if (count($parts) >= 2) {
+            return sprintf('%s.%s', $parts[0], $parts[1]);
+        }
+
+        return $this->generateSigninInput();
```

Comment:
hey @jshayes thanks for this! Any reason we would need to fallback to the original `generateSigninInput`?